### PR TITLE
[OB-4761] fix: unreal crashes on oko asset drag and drop without internet connection

### PR DESCRIPTION
**Pull request by MAG-ChristopherAtkinson**
This is quite a complicated problem; the PR here resolves the symptom but not the underlying cause. I will capture the overarching issues externally to this, but here is some additional information to highlight the main problem at a high level:

 - CSP, when experiencing an unexpected network interruption, will only receive a notification when attempting to operate. 
 - POCO will then send `ConnectionResetException` "Connection reset by peer" exception
   - POCO does support a `ConnectionAbortedException`, but it seems that the expected exception in an unexpected network interruption is `ConnectionResetException`
   - In any case, we trigger the same behaviour on all failure cases in `CSPWebSocketClientPOCO::HandleReceiveError`
 - The exceptions thrown do not always interrupt the current operation, meaning that we execute methods such as 'SendObjectMessage' with invalid data while in a disconnected state (e.g. anytime we require external data as part of the request, 'GenerateObjectIds')
 - Resulting in undefined behaviour from POCO when attempting to re-construct `websocket_factory` (Leaning towards a segmentation fault)

In the create entity multiplayer flow, we see this exact issue, where we invoke 'SendObjectMessage' event with an invalid ID due to a previous failed 'GenerateObjectIds' event. 

> [!NOTE]
> This is where the fix for the symptom is implemented, preventing execution of subsequent calls. I did a quick audit over similar functions to see if we had other potential edge cases, to no avail.

### CODE REVIEW

**Comment by MAG-ElliotMorris** [comment]
[Question] Is it doable to have a test that fails before this change, and passes now?

I understand the internet connection bit might make this hard, not sure how developed our mocks are in this area.

File: Tests/src/PublicAPITests/OnlineRealtimeEngineTests.cpp [hunk 1]
```diff
@@ -281,3 +285,42 @@ CSP_PUBLIC_TEST_WITH_MOCKS(CSPEngine, OnlineRealtimeEngineTests, TestErrorLogged
...
+            });
+
+    using MockEntityCreatedCallback = testing::MockFunction<void(SpaceEntity*)>;
+    MockEntityCreatedCallback MockCallback;
+
+    // Expect the callback gets nullptr (not the greatest error return...)
+    EXPECT_CALL(MockCallback, Call(nullptr));
+
+    // Expect that we log the error message once and only once
+    const csp::common::String ErrorMsg = "Failed to generate object ID.";
```
Comment (MAG-ChristopherAtkinson) [inline Tests/src/PublicAPITests/OnlineRealtimeEngineTests.cpp:319]: In the event that the function failed to return when an exception is thrown, this would trigger multiple entries failing the test

File: Tests/src/PublicAPITests/OnlineRealtimeEngineTests.cpp [hunk 2]
```diff
@@ -70,7 +70,8 @@ CSP_PUBLIC_TEST_WITH_MOCKS(CSPEngine, OnlineRealtimeEngineTests, TestSuccessInRe
                 EXPECT_FALSE(Result.get_exception());
                 EXPECT_EQ(Result.get(), uint64_t(55));
             })
-        .then(async::inline_scheduler(), [](async::task<void> CheckForErrorsTask)
+        .then(async::inline_scheduler(),
```
Comment (MAG-ChristopherAtkinson) [inline Tests/src/PublicAPITests/OnlineRealtimeEngineTests.cpp:73]: Apologies for the noise. It seems the file was formatted during my changes

File: Tests/src/PublicAPITests/OnlineRealtimeEngineTests.cpp [hunk 3]
```diff
@@ -281,3 +285,42 @@ CSP_PUBLIC_TEST_WITH_MOCKS(CSPEngine, OnlineRealtimeEngineTests, TestErrorLogged
...
+
+                Callback(Value, ExceptionPtr);
+
+                return async::make_task(std::make_tuple(Value, ExceptionPtr));
+            });
+
+    using MockEntityCreatedCallback = testing::MockFunction<void(SpaceEntity*)>;
+    MockEntityCreatedCallback MockCallback;
+
+    // Expect the callback gets nullptr (not the greatest error return...)
```
Comment (MAG-ElliotMorris) [inline Tests/src/PublicAPITests/OnlineRealtimeEngineTests.cpp:315]: [Comment] You can say that again.
